gymnasium/spaces/box.py

```diff
@@ -335,14 +328,26 @@ def _broadcast(
 ) -> NDArray[Any]:
     """Handle infinite bounds and broadcast at the same time if needed."""
     if is_float_integer(value):
-        value = get_inf(dtype, np.sign(value)) if np.isinf(value) else value
+        if np.isinf(value) and np.dtype(dtype).kind == "i":
+            value = (
+                get_integer_infinite(dtype, np.sign(value))
+                if np.isinf(value)
+                else value
+            )
+
         value = np.full(shape, value, dtype=dtype)
-    else:
-        assert isinstance(value, np.ndarray)
-        if np.any(np.isinf(value)):
+
+    elif isinstance(value, np.ndarray):
+        if np.any(np.isinf(value)) and np.dtype(dtype).kind == "i":
             # create new array with dtype, but maintain old one to preserve np.inf
-            temp = value.astype(dtype)
-            temp[np.isinf(value) & (value > 0.0)] = get_inf(dtype, 1.0)
-            temp[np.isinf(value) & (value < 0.0)] = get_inf(dtype, -1.0)
-            value = temp
+            bounded_value = value.astype(dtype)
+            bounded_value[np.isposinf(value)] = get_integer_infinite(dtype, 1.0)
```

Comment:
Can we not just set `np.iinfo(dtype).max - 2` and same with line below, this is more clear that is happening

Reply:
Good point.